test/kafka-e2e/kafka_test.go

```diff
@@ -42,3 +42,34 @@ func TestKafka(t *testing.T) {
 	DeletePipelines()
 	WaitForPodsToBeDeleted()
 }
+
+func TestKafkaAsync(t *testing.T) {
+	defer Setup(t)()
+
+	topic := CreateKafkaTopic()
+	sinkTopic := CreateKafkaTopic()
+
+	CreatePipeline(Pipeline{
+		ObjectMeta: metav1.ObjectMeta{Name: "kafka"},
+		Spec: PipelineSpec{
+			Steps: []StepSpec{{
+				Name:    "main",
+				Cat:     &Cat{},
+				Sources: []Source{{Kafka: &KafkaSource{Kafka: Kafka{Topic: topic}}}},
+				Sinks:   []Sink{{Kafka: &KafkaSink{Kafka: Kafka{Topic: sinkTopic}, Async: true}}},
+			}},
+		},
+	})
+	WaitForPipeline()
+	WaitForPod()
+
+	PumpKafkaTopic(topic, 17)
+
+	WaitForPipeline(UntilMessagesSunk)
+
+	WaitForStep(TotalSourceMessages(17))
+	WaitForStep(TotalSunkMessages(17))
+
+	DeletePipelines()
```

Comment:
I think you need to use `ExpectMetrics` to verify the metrics

Reply:
added, `2021/08/20 14:39:08 expect metric "sinks_kafka_produced_successes" to be 17.000000 on 3569`